modules/gatsby/src/gatsby-node/create-pages.js

```diff
@@ -164,12 +175,57 @@ function createDocPages({ graphql, actions }) {
         categorySet.add(edge.node.frontmatter.category);
       }
 
+      const relativeLinks = pathToSlug.reduce((prev, { source, target }) => {
```

Comment:
Total Nit: Not a big fan of `reduce`, prefer `for of` or `forEach`. Though theoretically elegant, I find `reduce` code harder to read.

More importantly this many lines deserves to be put in their own function.